ui/app/components/app/account-menu/account-menu.component.js

```diff
@@ -195,7 +195,7 @@ export default class AccountMenu extends Component {
     const { t } = this.context
     // Any account that's not from the HD wallet Keyring can be removed
     const { type } = keyring
-    const isRemovable = type !== 'HD Key Tree'
+    const isRemovable = !(type === 'HD Key Tree' || type === 'TrustVault')
```

Comment:
Nit: this suggestion here might be a cleaner diff

```suggestion
    const isRemovable = type !== 'HD Key Tree' && type !== 'TrustVault'
```